test/decl/enum/bool_raw_value.swift

```diff
@@ -0,0 +1,16 @@
+// RUN: %target-typecheck-verify-swift
+extension Bool: ExpressibleByIntegerLiteral {
+	public init(integerLiteral value: Int) {
+		self = value != 0
+	}
+}
+
+enum IsDefinitelyRecursive : Bool, Equatable, Hashable {
+  case recursive=false
+}
+
+// expected-error@+1{{'IsRecursive' declares raw type 'Bool', but does not conform to RawRepresentable and conformance could not be synthesized}}
+enum IsRecursive : Bool, Equatable, Hashable {
```

Comment:
Some more tests: an @objc enum (and if we allow this, an IRGen test), an enum with an invalid raw value expression.

Reply:
`@objc` enums are banned; I've added more tests